cli/cook/cli.py

```diff
@@ -93,10 +94,16 @@ def run(args, plugins):
             defaults = config_map.get('defaults')
             action_defaults = (defaults.get(action) if defaults else None) or {}
             # load_target_clusters needs to be aware if there are any action-specific cluster defaults
+            # We want to treat cluster & pool as a pair here, so if either is specified on the cli,
+            # we will override the defaults.
+            # Additionally, we must `pop` to remove the disallowed 'cluster' key, regardless of
+            # whether we use the action_defaults or the cli flags
             if action_defaults:
-                # coalesce the defaults with cli flags overriding action config defaults
-                # using `pop` to remove the disallowed 'cluster' key
-                cluster = cluster or action_defaults.pop("cluster", None)
+                action_default_cluster = action_defaults.pop("cluster", None)
+                if cluster or pool_name:
+                    action_defaults.pop("pool-name", None)
+                else:
+                    cluster = action_default_cluster
```

Comment:
This is the new logic that allows us to override the cluster and pool defaults if either cluster or pool cli flags are specified